doc/advanced_topics.md

```diff
@@ -110,3 +110,118 @@ To implement your own transient event store, the only requirement is to set the
  - Run pre-commit manually
 
     `$ pre-commit run --all-files`
+
+## Retrieving the EDM definition from a data file
+It is possible to get the EDM definition(s) that was used to generate the
+datatypes that are stored in a data file. This makes it possible to re-generate
+the necessary code and build all libraries again in case they are not easily
+available otherwise. To see which EDM definitions are available in a data file
+use the `podio-dump` utility
+
+```bash
+podio-dump <data-file>
+```
+which will give an (exemplary) output like this
+```
+input file: <data-file>
+
+EDM model definitions stored in this file: edm4hep
+
+[...]
+```
+
+To actually dump the model definition to stdout use the `--dump-edm` option
+and the name of the datamodel you want to dump:
+
+```bash
+podio-dump --dump-edm edm4hep <data-file> > dumped_edm4hep.yaml
+```
+
+Here we directly redirected the output to a yaml file that can then again be
+used by the `podio_class_generator.py` to generate the corresponding c++ code
+(or be passed to the cmake macros).
+
+**Note that the dumped EDM definition is equivalent but not necessarily exactly
+the same as the original EDM definition.** E.g. all the datatypes will have all
+their fields (`Members`, `OneToOneRelations`, `OneToManyRelations`,
+`VectorMembers`) defined, and defaulted to empty lists in case they were not
+present in the original EDM definition. The reason for this is that the embedded
+EDM definition is the pre-processed and validated one [as described
+below](#technical-details-on-edm-definition-embedding)
+
+### Accessing the EDM definition programmatically
+The EDM definition can also be accessed programmatically via the
+`[ROOT|SIO]FrameReader::getEDMDefinition` method. It takes an EDM name as its
```

Comment:
We have to be a bit more careful in how we name things. There is the EDM (the definition of all data you ever want to store) and the definition of the individual data types. I think we should clean up the interfaces to talk about datatype definitions and not EDM definitions if we are talking about single types / collections. That need for check holds for all what contains `EDM` at the moment.